python/strings_udf/strings_udf/_typing.py

```diff
@@ -23,19 +24,33 @@
 
 
 # String object definitions
-class DString(types.Type):
+class UDFString(types.Type):
+
+    np_dtype = np.dtype("object")
+
     def __init__(self):
-        super().__init__(name="dstring")
+        super().__init__(name="udf_string")
         llty = default_manager[self].get_value_type()
```

Comment:
Out of scope for this PR, but one thing that would help me follow a lot of our numba code and might also help others is using more fully delineated names for variables involving "types". I remember it took me multiple passes the first time that I read through any of our JIT code because I couldn't figure out what `retty` meant, and I still do a double-take. Those kinds of variable names are common enough that it would probably meaningfully help readability.

Reply:
I agree with this yeah. Just a habit picked up from the numba codebase 😅

Reply:
Super agree. For me, the naming is the single hardest part of understanding the code that interacts with Numba. I still struggle with it.